packages/@sanity/form-builder/src/sanity/DefaultAssetSource/AssetThumb.tsx

```diff
@@ -0,0 +1,178 @@
+import type {Subscription} from 'rxjs'
+import React, {useState, useEffect, useRef} from 'react'
+import styled from 'styled-components'
+import {Button, useToast} from '@sanity/ui'
+import {Asset as AssetType} from '@sanity/types'
+import {FullscreenSpinner} from '../../components/FullscreenSpinner'
+import {versionedClient} from '../versionedClient'
+import {Checkerboard} from '../../components/Checkerboard'
+import {AssetUsageDialog} from './AssetUsageDialog'
+import {AssetMenu} from './AssetMenu'
+import {AssetMenuAction} from './types'
+
+interface AssetProps {
+  asset?: AssetType
+  isSelected: boolean
+  onClick?: (...args: any[]) => any
+  onKeyPress?: (...args: any[]) => any
+  onDeleteFinished: (...args: any[]) => any
+}
+
+// Get pixel density of the current device
+const DPI =
+  typeof window === 'undefined' || !window.devicePixelRatio
+    ? 1
+    : Math.round(window.devicePixelRatio)
+
+const Image = styled.img`
+  position: absolute;
+  z-index: 1;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  display: block;
+  object-fit: contain;
+`
+
+const Container = styled(Checkerboard)`
+  position: relative;
+  z-index: 1;
+  padding-bottom: 100%;
+`
+
+const Root = styled.div`
+  position: relative;
+  display: inherit;
+`
+
+const MenuContainer = styled.div`
+  box-sizing: border-box;
+  position: absolute;
+  z-index: 2;
+  display: none;
+  top: 3px;
+  right: 3px;
+
+  ${Root}:hover & {
+    display: block;
+  }
+`
+
+const AssetThumb = (props: AssetProps) => {
+  const toast = useToast()
+  const deleteRef$ = useRef<Subscription>()
+  const {asset, onClick, onKeyPress, onDeleteFinished, isSelected} = props
+  const [showUsageDialog, setShowUsageDialog] = useState(false)
+  const [showDeleteDialog, setShowDeleteDialog] = useState(false)
+  const [isDeleting, setIsDeleting] = useState(false)
+
+  useEffect(() => {
+    return () => {
+      if (deleteRef$.current) {
+        deleteRef$.current.unsubscribe()
+      }
+    }
+  }, [])
+
+  const handleConfirmDelete = () => {
+    setShowDeleteDialog(true)
+  }
+
+  const handleDialogClose = () => {
+    setShowUsageDialog(false)
+    setShowDeleteDialog(false)
+  }
+
+  const handleToggleUsageDialog = () => {
+    setShowUsageDialog(true)
+  }
+
+  const handleDeleteError = (error) => {
+    toast.push({
+      closable: true,
+      status: 'error',
+      title: 'Image could not be deleted',
```

Comment:
maybe I'm missing something, but based on the file name this component seems to handle both images and files - should this message reflect that and say _Asset could not be deleted_ instead?

Reply:
This is only used for images, as you can see in https://github.com/sanity-io/sanity/blob/90228384c3ab61492beb48fa554910eec3ea6c84/packages/%40sanity/form-builder/src/sanity/DefaultAssetSource/DefaultSource.tsx#L127-L159 - rendered conditionally here https://github.com/sanity-io/sanity/blob/90228384c3ab61492beb48fa554910eec3ea6c84/packages/%40sanity/form-builder/src/sanity/DefaultAssetSource/DefaultSource.tsx#L189